x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/saml/RestSamlCompleteLogoutAction.java

```diff
@@ -47,10 +47,12 @@
         PARSER = new ObjectParser<>("saml_complete_logout", SamlCompleteLogoutRequest::new);
 
     static {
-        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setQueryString, new ParseField("queryString"));
+        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setQueryString,
+            new ParseField("queryString").withAllDeprecated("query_string"));
         PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setContent, new ParseField("content"));
         PARSER.declareStringArray(SamlCompleteLogoutRequest::setValidRequestIds, new ParseField("ids"));
         PARSER.declareString(SamlCompleteLogoutRequest::setRealm, new ParseField("realm"));
+        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setQueryString, new ParseField("query_string"));
```

Comment:
This seems weird. Can't you just use the deprecated names option on a single field?
```suggestion
        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setQueryString, new ParseField("query_string", "queryString"));
        PARSER.declareStringOrNull(SamlCompleteLogoutRequest::setContent, new ParseField("content"));
        PARSER.declareStringArray(SamlCompleteLogoutRequest::setValidRequestIds, new ParseField("ids"));
        PARSER.declareString(SamlCompleteLogoutRequest::setRealm, new ParseField("realm"));
```

Reply:
👍